app/metrics/contentful.py

```diff
@@ -79,23 +90,65 @@ def update_entry_using_json_response(content_type, id, data):
         'X-Contentful-Content-Type': str(content_type),
         'X-Contentful-Version': str(version)
     }
-    
+
     response = requests.put(
         headers=hed,
         url=url,
         json=data
     )
     return response.json()
 
+
 def publish_entry(id, version):
     url = f'https://{Config.CTF_CMA_API_HOST}/spaces/{Config.CTF_SPACE_ID}/environments/master/entries/{id}/published'
     hed = {
         'Authorization': 'Bearer ' + Config.CTF_CMA_ACCESS_TOKEN,
         'X-Contentful-Version': str(version)
     }
-    
+
     response = requests.put(
         headers=hed,
         url=url
     )
     return response.json()
+
+
+def _have_featured_datasets(result):
+    if len(result['items'][0]) == 1:
+        featured_data = result['items'][0]
+        return 'featuredDatasets' in featured_data['fields'] and 'dateToClearFeaturedDatasets' in featured_data['fields']
+
+    return False
+
+
+def get_featured_datasets():
+    url = f'https://{Config.CTF_CDA_API_HOST}/spaces/{Config.CTF_SPACE_ID}/environments/master/entries'
+    hed = {
+        'Authorization': 'Bearer ' + Config.CTF_CDA_ACCESS_TOKEN,
+    }
+    q = {
+        'content_type': 'homepage',
+        'select': 'fields.featuredDatasets,fields.dateToClearFeaturedDatasets',
+    }
+
+    response = requests.get(
+        headers=hed,
+        params=q,
+        url=url
+    )
+
+    json_data = response.json()
+
+    featured_datasets = []
+    if _have_featured_datasets(json_data):
```

Comment:
Will it be better to wrap this in a try block, in case the response is not valid.